src/parser/parse.ts

```diff
@@ -469,6 +475,73 @@ function parseGroupOpen(token: GroupOpenToken, context: Context, state: State):
   return node;
 }
 
+function parseNamedCallout({kind, name, tag, args}: NamedCalloutToken): NamedCalloutNode {
+  const argumentValues: Array<string | number> =
+    args ?
+      args.split(',')
+        .filter((argument: string): boolean => argument !== '') // oniguruma skips over/ignores redundant/unnecessary commas
+        .map((argument: string): string | number => /^[+-]?\d+$/.exec(argument) ? +argument : argument)
+      : [];
+  const argument0 = argumentValues[0];
+  const argument1 = argumentValues[1];
+  const argument2 = argumentValues[2];
+  switch (name) {
+    case 'FAIL':
+    case 'MISMATCH':
+    case 'SKIP':
+      if (argumentValues.length > 0) {
+        throw new Error(`Named callout arguments not allowed "${argumentValues}"`);
+      }
+      break;
+    case 'ERROR': {
+      if (argumentValues.length > 1) {
+        throw new Error(`Named callout only one argument allowed "${argumentValues}"`);
+      }
+      if (typeof argument0 === 'string') {
+        throw new Error(`Named callout argument must be a number "${argument0}"`);
+      }
+    }
+      break;
+    case 'MAX':
+      if (argumentValues.length < 1 || argumentValues.length > 2) {
+        throw new Error(`Named callout must have one or two arguments "${argumentValues}"`);
+      }
+      if (typeof argument0 === 'string' && !/^[A-Za-z_]\w*$/.exec(argument0)) {
+        throw new Error(`Named callout argument one must be a number or tag "${argument0}"`);
+      }
+      if (argumentValues.length == 2 && typeof argument1 !== 'number' && !/^[<>X]$/.exec(argument1)) { // argument optional
+        throw new Error(`Named callout argument two must be a '<', '>', 'X' or a number "${argument1}"`);
+      }
+      break;
+    case 'COUNT':
+    case 'TOTAL_COUNT':
+      if (argumentValues.length > 1) {
+        throw new Error(`Named callout only one argument allowed "${argumentValues}"`);
+      }
+      if (argumentValues.length == 1 && typeof argument0 !== 'number' && !/^[<>X]$/.exec(argument0)) { // argument optional
+        throw new Error(`Named callout argument must be '<', '>', 'X' or a number "${argument0}"`);
+      }
+      break;
+    case 'CMP':
+      if (argumentValues.length !== 3) {
+        throw new Error(`Named callout must have three arguments "${argumentValues}"`);
+      }
+      if (typeof argument0 === 'string' && !/^[A-Za-z_]\w*$/.exec(argument0)) {
+        throw new Error(`Named callout argument one must be a tag or number "${argument0}"`);
+      }
+      if (typeof argument1 === 'number' || !/^(?:[<>!=]=|[<>])$/.exec(argument1)) {
```

Comment:
- `exec` -> `test`
- In this case it's probably better/clearer to use `!== 'string'` instead of `=== 'number'`, since we don't want the regex test to operate on it unless it's a string.